### PR TITLE
Update to euclid 0.8

### PR DESCRIPTION
**Pull request by mrobinson**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because it is covered by existing tests

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12809)

### CODE REVIEW

File: ports/cef/Cargo.lock [hunk 1]
```diff
@@ -580,6 +581,18 @@ dependencies = [
 ]
 
 [[package]]
+name = "euclid"
```
Comment (emilio) [inline ports/cef/Cargo.lock]: Hmmm... Seems like we're still pulling euclid 0.7 from CEF, that'd need to be fixed

**Comment by emilio** [comment]
LGTM with the CEF fix, and the confirmation that matrix operations don't need to change.

Thanks for doing this! :)

**Comment by mrobinson** [comment]
@emilio Thanks. I believe I've fixed the cef issue now. Matrix operations shouldn't need to change until euclid 0.9.

File: components/style/properties/longhand/box.mako.rs [hunk 2]
```diff
@@ -368,47 +368,42 @@ ${helpers.single_keyword("overflow-x", "visible hidden scroll auto",
...
-    });
-    static EASE_IN_OUT: TransitionTimingFunction =
-        TransitionTimingFunction::CubicBezier(Point2D {
-            x: 0.42,
-            y: 0.0,
-        }, Point2D {
-            x: 0.58,
-            y: 1.0,
-        });
+    #[inline(always)]
```
Comment (emilio) [inline components/style/properties/longhand/box.mako.rs]: Oh, also, sorry for not suggesting this before, but could you add here a `FIXME: Use static + const fn when available` or similar?

**Comment by emilio** [comment]
Sorry for that last-minute nit :/

r=me with that.

@bors-servo: delegate+

**Comment by jdm** [comment]
@bors-servo: r-
Both CI systems are unhappy.

**Comment by jdm** [comment]
`./mach test-unit` fails, and the geckolib Cargo.lock needs updating as well.

**Comment by mrobinson** [comment]
Ugh. Really sorry about that. I will try to have this done today. ☹

**Comment by larsbergstrom** [comment]
@bors-servo retry
- There was a minor update to servo-fontconfig-sys that should fix this build error.

**Comment by nox** [comment]
Superseded by #12817.

**Pull request closed without merging**